File: 백준/Gold/3109. 빵집/빵집.py
input = open(0).readline

R, C = map(int, input().split())
board = [list(input().rstrip()) for _ in range(R)]


def out_of_bound(x: int, y: int) -> bool:
    return not(0 <= x < R and 0 <= y < C)


def dfs(x: int, y: int) -> int:
    if out_of_bound(x, y) or board[x][y] == 'x':
        return 0
    elif y == C - 1:
        return 1
    else:
        board[x][y] = 'x'
        return dfs(x - 1, y + 1) or dfs(x, y + 1) or dfs(x + 1, y + 1)
        
        
print(sum(dfs(i, 0) for i in range(R)))

File: 백준/Gold/3109. 빵집/빵집.cc
#include <cstdio>

int R, C;
char board[10000][500];

bool dfs(int x, int y) {
    if(x < 0 || x >= R || y < 0 || y >= C || board[x][y] == 'x') {
        return false;
    } else if(y == C - 1) {
        return true;
    } else {
        board[x][y] = 'x';
        return dfs(x - 1, y + 1) || dfs(x, y + 1) || dfs(x + 1, y + 1);
    }
}

int main() {
    scanf(" %d %d", &R, &C);
    for(int r = 0; r < R; ++r) {
        for(int c = 0; c < C; ++c) {
            scanf(" %c", &board[r][c]);
        }
    }
    
    int ans = 0;
    for(int r = 0; r < R; ++r) {
        ans += dfs(r, 0);
    }
    printf("%d", ans);
    
    return 0;
}